wrangler-core/src/main/java/co/cask/directives/parser/ParseSimpleDate.java

```diff
@@ -78,16 +80,23 @@ public void destroy() {
         Object object = row.getValue(idx);
         // If the data in the cell is null or is already of date format, then
         // continue to next row.
-        if (object == null || object instanceof Date) {
+        if (object == null || object instanceof ZonedDateTime) {
           continue;
         }
         if (object instanceof String) {
           try {
-            Date date = format.parse((String) object);
-            row.setValue(idx, date);
+            // This implementation first creates Date object and then converts it into ZonedDateTime.
+            // DateTimeFormat java 8 api to apply pattern on a string to get ZonedDateTime, throws DateTimeException
```

Comment:
I don't understand what this comment means.

Reply:
Basically to create `ZonedDateTime`,  the pattern/object to be parsed must have time and timezone components. If they are absent in case of pattern `yyyy-mm-dd`, `DateTimeParseException` is thrown.
For example:
```java
@Test
  public void testDate() {
    DateTimeFormatter destinationFmt = DateTimeFormatter.ofPattern("yyyy-mm-dd");
    ZonedDateTime parse = ZonedDateTime.parse("2018-12-21", destinationFmt);
  }
```
Above code throws `DateTimeParseException`. 

I have rephrased the comment with an example.